autolens/pipeline/phase/interferometer/analysis.py

```diff
@@ -195,52 +189,36 @@ def visualize(self, paths: af.Paths, instance, during_analysis):
             tracer=tracer, hyper_background_noise=hyper_background_noise
         )
 
-        if tracer.has_mass_profile:
-
-            try:
-
-                visualizer = self.visualizer.new_visualizer_with_preloaded_critical_curves_and_caustics(
-                    preloaded_critical_curves=tracer.critical_curves,
-                    preloaded_caustics=tracer.caustics,
-                )
-
-            except (Exception, IndexError, ValueError):
-
-                visualizer = self.visualizer
-
-        else:
-
-            visualizer = self.visualizer
-
-        visualizer.visualize_ray_tracing(
-            paths=paths, tracer=fit.tracer, during_analysis=during_analysis
+        visualizer = vis.Visualizer(visualize_path=paths.image_path)
+        visualizer.visualize_interferometer(
+            interferometer=self.masked_interferometer.interferometer
+        )
+        visualizer.visualize_fit_interferometer(
+            fit=fit, during_analysis=during_analysis
+        )
+        visualizer.visualize_tracer(
+            tracer=fit.tracer, grid=fit.grid, during_analysis=during_analysis
         )
-        visualizer.visualize_fit(paths=paths, fit=fit, during_analysis=during_analysis)
+        if fit.inversion is not None:
+            visualizer.visualize_inversion(
+                inversion=fit.inversion, during_analysis=during_analysis
+            )
 
-        self.visualizer.visualize_hyper_images(
-            paths=paths,
+        visualizer.visualize_hyper_images(
             hyper_galaxy_image_path_dict=self.hyper_galaxy_image_path_dict,
             hyper_model_image=self.hyper_model_image,
-            contribution_maps_of_galaxies=tracer.contribution_maps_of_planes,
+            tracer=tracer,
         )
 
-        if self.visualizer.plot_fit_no_hyper:
+        if visualizer.plot_fit_no_hyper:
 
             fit = self.masked_interferometer_fit_for_tracer(
-                tracer=tracer,
-                hyper_background_noise=hyper_background_noise,
-                use_hyper_scalings=False,
+                tracer=tracer, hyper_background_noise=None, use_hyper_scalings=False

```

Comment:
There's loads of places I'd put arguments on new lines instead of one line. Migth be worth looking up black configuration :P

Reply:
You might be right...